meson.build

```diff
@@ -0,0 +1,544 @@
+elif libpoppler_dep.found() and cairo_dep.found()
+    if get_option('poppler-module') and modules_enabled
+        cfg_var.set('POPPLER_MODULE', '1')
+    else
+        libvips_deps += libpoppler_dep
+        libvips_deps += cairo_dep
+    endif
+    cfg_var.set('HAVE_POPPLER', '1')
+endif
+
+libnifti_dep = cc.find_library('niftiio', has_headers: 'nifti1_io.h', required: false)
+if not libnifti_dep.found()
+    libnifti_dep = dependency('NIFTI', method : 'cmake', modules : ['NIFTI'], required: get_option('nifti'))
+endif
+if libnifti_dep.found()
+    libvips_deps += libnifti_dep
+    cfg_var.set('HAVE_NIFTI', '1')
+endif
+
+if cc.has_header('sys/file.h')
+    cfg_var.set('HAVE_SYS_FILE_H', '1')
+endif
+if cc.has_header('sys/param.h')
+    cfg_var.set('HAVE_SYS_PARAM_H', '1')
+endif
+if cc.has_header('sys/mman.h')
+    cfg_var.set('HAVE_SYS_MMAN_H', '1')
+endif
+if cc.has_header('unistd.h')
+    cfg_var.set('HAVE_UNISTD_H', '1')
+endif
+if cc.has_header('io.h')
+    cfg_var.set('HAVE_IO_H', '1')
+endif
+if cc.has_header('direct.h')
+    cfg_var.set('HAVE_DIRECT_H', '1')
+endif
+if cc.has_header('windows.h')
+    cfg_var.set('HAVE_WINDOWS_H', '1')
+endif
+if get_option('deprecated')
+    cfg_var.set('ENABLE_DEPRECATED', '1')
+endif
+if get_option('nsgif')
+    cfg_var.set('HAVE_NSGIF', '1')
+endif
+if get_option('ppm')
+    cfg_var.set('HAVE_PPM', '1')
+endif
+if get_option('analyze')
+    cfg_var.set('HAVE_ANALYZE', '1')
+endif
+if get_option('radiance')
+    cfg_var.set('HAVE_RADIANCE', '1')
+endif
+
+gettext_domain = 'vips@0@.@1@'.format(version_major, version_minor)
+cfg_var.set_quoted('GETTEXT_PACKAGE', gettext_domain)
+cfg_var.set_quoted('VIPS_PREFIX', prefix_dir)
+cfg_var.set_quoted('VIPS_LIBDIR', lib_dir)
+if cc.has_function('ngettext')
+    cfg_var.set('ENABLE_NLS', 1)
+else
+    libintl_dep = cc.find_library('intl', required : false)
+    if libintl_dep.found()
+        libvips_deps += libintl_dep
+        cfg_var.set('ENABLE_NLS', 1)
+    endif
+endif
+
+target_system = target_machine.system()
+if target_system == 'darwin'
+    profile_dir = '/Library/ColorSync/Profiles'
+elif target_system == 'windows'
+    # need double escapes since this will get pasted into a #define in a C
+    # header ... the C:\Windows is usually overwritten with the result of
+    # GetWindowsDirectoryW()
+    profile_dir = 'C:\\\\Windows\\\\System32\\\\spool\\\\drivers\\\\color'
+else
+    profile_dir = get_option('prefix') / get_option('datadir') / 'color' / 'icc'
+endif
+cfg_var.set_quoted('VIPS_ICC_DIR', profile_dir)
+
+config_file = configure_file(
+    configuration: cfg_var,
+    output: 'config.h'
+)
+
+config_dep = declare_dependency(
+    sources: config_file,
+    include_directories: include_directories('.'),
+    compile_args: '-DHAVE_CONFIG_H=1',
+)
+
+libvips_deps += config_dep
+
+subdir('libvips')
+if get_option('gtk_doc')
+    subdir('doc')
+endif
+subdir('cplusplus')
+subdir('man')
```

Comment:
Would it be possible to wrap each of these last 5 `subdir` directives ("man" through to "fuzz") with their own boolean options, defaulting to true, to allow building only the shared/static libraries without tools and tests? This is based on a discussion at https://github.com/lovell/sharp-libvips/issues/127#issuecomment-1030901615

Reply:
I don't really see the point e.g. "man" just creates installation rules which have no build time effect, only an install-time effect. You can use `meson install --tags runtime` to avoid installing any files tagged as `man`, see https://mesonbuild.com/Installing.html#installation-tags for details.

For the other directories, they actually do create build rules, but those could for the most part be defined with `build_by_default: false` if you don't want to make everyone automatically build them (they could be built by name if desired).

Granted Meson currently doesn't respect build_by_default for test executables, ideally they should not be built by `ninja all` in that case but still get built by e.g. `ninja test-prereq` and pulled in as a dependency of `ninja test`. However, this theoretical `test-prereq` is instead merged into `all` and that's a bit annoying. We do intend to fix it, there's a WIP PR for it somewhere... in the meantime many people do have *one* project option "tests=false" which disables descending into `tests/meson.build` or what have you.

Adding 5 different options seems like a lot of cognitive overhead for very little gain.

Reply:
Thank you, I agree, options fatigue is a real thing. Meson's use of a human-readable `meson_options.txt` file and its ability to fail a build when it sees unknown options is a huge improvement. I wasn't aware of the predefined `--tags`, very useful, thanks. I'd be happy with a single `tests` boolean option, which I guess would prevent descent into the `test` and `fuzz` subdirs.